pep-0484.txt

```diff
@@ -584,7 +585,6 @@ argument(s) is substituted.  Otherwise, ``Any`` is assumed.  Example::
   class Node(Generic[T]):
       def __init__(self, label: T = None) -> None:
           ...
-      x = None  # Type: T
```

Comment:
Sorry, maybe I miss something. Why is this line removed?

Reply:
There was a typo here, `Type` should be `type`. But the line is relevant, it is used below on lines 620+ (in the original numbering).

Reply:
@Mariatta It is my mistake!


Maybe I'm wrong again but maybe it's better to write it like this:
```
class Node(Generic[T]):
    x = None # type: T
    def __init__(self, label: T=None) -> None:
        ...
```

- it is visually clear that the `x` attribute  is created purposefully, that this is not a typo (yes, it is visible if you are attentive and read to line 620 too);
- the `type` writing in lowercase (does not associate with typing.Type).


*For me, it's still a mystery why `x = None` of the `T` type ... maybe:*
```
class Node(Generic[T]):
    x = None # type: T
    def __init__(self, label: T=None, x: T=None) -> None:
        self.x = x
        ...
```

*So, I took more than I could carry away ...*

Reply:
- It's fine to move the definition of `x` before `__init__`.
- The capitalization of `Type` was a typo in the original PEP 484.
- The initialization with `None` is valid because (until PEP 526, Python 3.6) there was no way to write a class attribute without an initialization. The PEP has an explicit exception for this case around line 1495.
- I recommend adding an extra comment to the definition of `x` pointing forward, e.g.
```py
    x = None  # type: T  # Instance attribute (see below)
```

Reply:
I totally agree with a proposal move the definition of `x` before `__init__` and with proposal to add an extra comment to the definition of `x`.

And, thank you for the explanation!

Reply:
Can you please update your PR?

Reply:
@gvanrossum Done 1aa4b27, I left your comment - it is exhaustive.
@Mariatta Sorry my commit 0b2806c dubbed yours dab1007.

Reply:
No worries :) LGTM.